src/main/java/org/sonar/plugins/csharp/CSharpSensor.java

```diff
@@ -218,19 +255,19 @@ private static String escapeXml(String str) {
     return ImmutableMap.copyOf(builder);
   }
 
-  private void importResults(SensorContext context) {
+  void importResults(SensorContext context, Path workDirectory) {
     // Note: the no-sonar rules must be imported before issues, otherwise the affected issues won't get excluded!
-    parseProtobuf(context, ProtobufImporters.metricsImporter(fileLinesContextFactory, noSonarFilter), METRICS_OUTPUT_PROTOBUF_NAME);
-    parseProtobuf(context, ProtobufImporters.issuesImporter(CSharpSonarRulesDefinition.REPOSITORY_KEY), ISSUES_OUTPUT_PROTOBUF_NAME);
-    parseProtobuf(context, ProtobufImporters.highlightImporter(), HIGHLIGHT_OUTPUT_PROTOBUF_NAME);
-    parseProtobuf(context, ProtobufImporters.symbolRefsImporter(), SYMBOLREFS_OUTPUT_PROTOBUF_NAME);
-    parseProtobuf(context, ProtobufImporters.cpdTokensImporter(), CPDTOKENS_OUTPUT_PROTOBUF_NAME);
+    parseProtobuf(context, ProtobufImporters.metricsImporter(fileLinesContextFactory, noSonarFilter), workDirectory, METRICS_OUTPUT_PROTOBUF_NAME);
+    parseProtobuf(context, ProtobufImporters.issuesImporter(CSharpSonarRulesDefinition.REPOSITORY_KEY), workDirectory, ISSUES_OUTPUT_PROTOBUF_NAME);
+    parseProtobuf(context, ProtobufImporters.highlightImporter(), workDirectory, HIGHLIGHT_OUTPUT_PROTOBUF_NAME);
+    parseProtobuf(context, ProtobufImporters.symbolRefsImporter(), workDirectory, SYMBOLREFS_OUTPUT_PROTOBUF_NAME);
+    parseProtobuf(context, ProtobufImporters.cpdTokensImporter(), workDirectory, CPDTOKENS_OUTPUT_PROTOBUF_NAME);
   }
 
-  private static void parseProtobuf(SensorContext context, ProtobufImporter importer, String filename) {
-    File protobuf = new File(toolOutput(context.fileSystem()), filename);
-    if (protobuf.isFile()) {
-      importer.accept(context, protobuf);
+  private static void parseProtobuf(SensorContext context, ProtobufImporter importer, Path workDirectory, String filename) {
+    Path protobuf = workDirectory.resolve(filename);
+    if (Files.exists(protobuf)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace this with a call to the "toFile().exists" method [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS3725)